Common/TableProducer/PID/pidTOFFull.cxx

```diff
@@ -35,15 +39,47 @@ using namespace o2::track;
 void customize(std::vector<o2::framework::ConfigParamSpec>& workflowOptions)
 {
   std::vector<ConfigParamSpec> options{{"add-qa", VariantType::Int, 0, {"Produce TOF PID QA histograms"}}};
+  options.push_back({"add-qa-ev-time", VariantType::Int, 0, {"Produce TOF PID QA histograms for TOF event time"}});
   std::swap(workflowOptions, options);
 }
 
 #include "Framework/runDataProcessing.h"
 
+namespace o2::aod
+{
+namespace tofeventtime
+{
+DECLARE_SOA_COLUMN(TOFEvTime, tofEvTime, float);       //! TOF event time
+DECLARE_SOA_COLUMN(TOFEvTimeErr, tofEvTimeErr, float); //! TOF event time error
+DECLARE_SOA_COLUMN(TOFEvTimeMult, tofEvTimeMult, int); //! TOF event time multiplicity
+} // namespace tofeventtime
+
+DECLARE_SOA_TABLE(TOFEvTime, "AOD", "TOFEvTime", //! Table of the TOF event time
+                  tofeventtime::TOFEvTime,
+                  tofeventtime::TOFEvTimeErr,
+                  tofeventtime::TOFEvTimeMult);
+} // namespace o2::aod
+
+template <typename trackType>
+bool filterForTOFEventTime(const trackType& tr)
+{
+  return (tr.hasTOF() && tr.p() > 0.5f && tr.p() < 2.f && tr.trackType() == o2::aod::track::TrackTypeEnum::Track);
```

Comment:
So this work only for Run 3 tracks. Is this correct?

Reply:
Yes, this is intended, one could have a separate filter for the Run2 but in principle TOF is already used for the T0 determination and the info is kept in the Run2 AOD